htlcswitch/switch.go

```diff
@@ -1932,6 +1940,18 @@ func (s *Switch) AddLink(link ChannelLink) error {
 		)
 	}
 
+	// Since the link addition has succeeded, this channel is now officially
+	// active and we can tell the ChannelNotifier about it.
+	peerPub := link.Peer().PubKey()
+	channel, err := s.cfg.DB.FetchOpenChannel(peerPub[:], link.ChannelPoint())
+	if err != nil {
+		log.Warnf("errored fetching channel with channelpoint=%v: %v",
+			link.ChannelPoint(), err)
+		return nil
+	}
+
+	s.cfg.NotifyActiveChannelEvent(channel)
```

Comment:
one design question here, is do we want to notify once the channel becomes active for forwarding? or just notify on whether or not it's in the switch. right now, this will notify on pending and live links

the distinction is that a pending link is still waiting for confirmation, but indicates that funding has completed successfully. it will not be usable until the transaction confirms sufficiently, and the peers have exchanged funding lock msgs. perhaps we want to make distinction in the notification api between a pending and a live link? we could then later send a live ntfn when `UpdateShortChanID` is called on the pending link